sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/ResourceDeleteCancelEventData.java

```diff
@@ -243,18 +296,56 @@ public ResourceDeleteCancelEventData setAuthorization(String authorization) {
      * Get the claims property: The properties of the claims.
      *
      * @return the claims value.
+     * @deprecated This method is no longer supported since v4.9.0.
+     * <p> Use {@link ResourceDeleteCancelEventData#getResourceClaims()} instead.
      */
+    @Deprecated
     public String getClaims() {
-        return this.claims;
+        final Map<String, String> resourceClaims = getResourceClaims();
+        if (!resourceClaims.isEmpty()) {
+            try {
+                return defaultSerializerAdapter.serialize(resourceClaims, SerializerEncoding.JSON);
```

Comment:
Add unit test for this as we should make sure the map is getting serialized and deserialized correctly.